### DISCUSSION TITLE
orcid.py pull duplicate publication when orchid api have both doi and eid record

### DISCUSSION THREAD

**Discussion by hiiiyilingzhang** [greenelab/lab-website-template#200] [Q&A] [answered]
Dear Vincent,

Thanks for active development! I encounter an issue when retrieve publication using v1.1.5.

You can find in [this page](https://aclab-bioinfo.github.io/aclab_website_1.1.5/publications/), 2022 panel. Some paper using `eid` shows on author info and wrong date. I checked the file from orcid api, it seems like when a publication have both doi and eid record, the `orcid.py` script would pull both record to the source. For example, if you search `Specific COX-2 inhibitor` in this page, it will have two records in doi and eid format separately. While v1.0.0 didn't have this problem.

It would be really helpful if you can take a look at this issue. Thanks a lot!!

Bests,
Elaine

**Comment by vincerubinetti** [comment, marked as answer]
It looks like you've changed something since you opened this discussion, because I'm not seeing a duplicate of `Specific COX-2 inhibitor` on that page. I just see one:

> [Specific COX-2 inhibitor, NS-398, suppresses cellular proliferation and induces apoptosis in human hepatocellular carcinoma cells](https://doi.org/f99srh)
Alfred Cheng, Henry Chan, Wai Leung, Nathalie Wong, Philip Johnson, Joseph Sung
International Journal of Oncology   ·   01 Jul 2003   ·   doi:10.3892/ijo.23.1.113

It seems like you [commented out the `orcid.py` script's ability to use ORCID citation details](https://github.com/ACLab-Bioinfo/aclab_website_1.1.5/commit/c1d6108498d33c6ee7a91fdc3f55868b5651a804). That is perfectly fine as a customization, but the script was behaving as intended.

The reason you had a duplicate before is because you have that [same paper's doi:10.3892/ijo.23.1.113 in your `sources.yaml`](https://github.com/ACLab-Bioinfo/aclab_website_1.1.5/blob/main/_data/sources.yaml). In fact, you have it twice, but the cite process removes the duplicate there. Then, your `- orcid: 0000-0003-2345-6951` was probably retrieving that same paper, but for whatever reason, the ORCID api didn't have the DOI and only had the EID.

Version 1.0.0 of the template would've just left the EID as a bare ID, but the newer versions actually try to keep some of the citation details (title/publisher/etc) that ORCID gives us, because Manubot can't give us anything for EIDs. And the template can't remove that duplicate because it's a different ID and it can't be sure it's the same paper.

The solution here is probably not to comment out those lines in `orcid.py`, but instead just remove `doi:10.3892/ijo.23.1.113` from your `sources.yaml`. 

When using ORCID, I recommend starting with _just_ the `orcid.yaml`, and no `sources.yaml` at all. Then, see what papers the ORCID lists, and use `sources.yaml` only to fill in the gaps as needed. Otherwise you end up with unexpected duplicates. I will put a note to this effect in the documentation.

Alternatively, you could leave `doi:10.3892/ijo.23.1.113` in `sources.yaml`, but then uncomment [this line in `cite.py`](https://github.com/greenelab/lab-website-template/blob/main/_cite/cite.py#L151C27-L151C27). This will basically remove a source completely if Manubot can't cite its ID. Since Manubot can cite most common ID types, this has the effect of removing more obscure ID types, like EID.

**Reply by hiiiyilingzhang** [reply]
Sorry for the late, I just found the reply. Thanks for your detailed explanation! I am working on it.
